src/base/Makefile

```diff
@@ -1,3 +1,16 @@
+# Whenever make is run in this directory, call ./get_version.sh script.
+# ./get_version.sh script regenerates ./version.h if necessary, e.g. if it
+# does not already exist or if the version number has changed.
+RET := $(shell ./get_version.sh; echo " $$?")
```

Comment:
could you please call this something different from RET, as RET could be confused for a return status?